pkg/kinflate/app/application.go

```diff
@@ -95,16 +95,32 @@ func (a *applicationImpl) Resources() (resource.ResourceCollection, error) {
 		return nil, errs
 	}
 
+	transRaw := reindexResourceCollection(raw)
+	allTransRes, err := resource.Merge(res, transRaw)
+	if err != nil {
+		return nil, err
+	}
+
 	allRes, err := resource.Merge(res, raw)
 	if err != nil {
 		return nil, err
 	}
 
+	ot, err := transformers.NewOverlayTransformer(patches)
+	if err != nil {
+		return nil, err
+	}
+	// Overlay transformer uses the index built using the original names.
+	err = ot.Transform(allRes)
+	if err != nil {
+		return nil, err
+	}
+
 	t, err := a.getTransformer(patches)
 	if err != nil {
 		return nil, err
 	}
-	err = t.Transform(allRes)
+	err = t.Transform(allTransRes)
```

Comment:
I'm confused at the end of this function. You created `allRes` and performed overly transformation on it. You also created `allTransRes` and performed other transformation on it. At the end, I expect to see some kind of merging or combining logic.

Reply:
Added some comments to explain it.
PTAL.

Reply:
Now I understand it. Looks good to me. I also verified the change against examples. Thank you for this fix.